### DISCUSSION TITLE
how to do IDENTITY_INSERT with lucid Factories on MSSQL?

### DISCUSSION THREAD

**Discussion by guntribam** [adonisjs/core#2779] [Help]
I'm having problems with this:
```ts
await Database.connection(connection).rawQuery(`SET IDENTITY_INSERT ${MyModel.table} ON`)

const result = await Factory.define(MyModel, (ctx) => ({}))
    .build()
    .merge(Object.values(enumType))
    .createMany(Object.keys(enumType).length)

await Database.connection(connection).rawQuery(`SET IDENTITY_INSERT ${MyModel.table} OFF`)
```
I get this error:
`insert into [MYMODEL] ([cAtividade], [iCodigo], [sDescricao]) output inserted.[iCodigo] values (@p0, @p1, @p2) - Cannot insert explicit value for identity column in table 'T_TIPO_PESSOA' when IDENTITY_INSERT is set to OFF.`

I did find this is a problem with knex [2262](https://github.com/knex/knex/issues/2262)

So i did this to solve the problem, and with it i lost the chance to use factories and automate the creation of these tables that only have "types" of something:

```ts
const rows = Object.values(enumType)
await Database.connection(myConnection).rawQuery(`
  SET IDENTITY_INSERT ${MyModel.table} ON

  INSERT INTO ${MyModel.table}
  (${Object.keys(rows[0])
    .map((key) => model.$columnsDefinitions.get(key)?.columnName)
    .join(',')})
  VALUES
  ${rows.map((row) => '(' + Object.values(row).map((v) => "'" + v + "'") + ')').join(',')}

  SET IDENTITY_INSERT ${MyModel.table} OFF
`)

```
There is a way to get the result of a Factory as sql to use it inside a rawQuery or something?

**Comment by guntribam** [comment]
@RomainLanz @thetutlage please help...

**Reply by thetutlage** [reply]
Hello! Nope, there is no way to get the result of the factory as a SQL. I read the Knex issue and looks like it is a deeper issue, so not sure if I can provide any help here
